riot/riot.py

```diff
@@ -567,19 +569,22 @@ def run(
         recreate_venvs: bool = False,
         out: t.TextIO = sys.stdout,
         pass_env: bool = False,
+        build: bool = False,
         cmdargs: t.Optional[t.Sequence[str]] = None,
         pythons: t.Optional[t.Set[Interpreter]] = None,
         skip_missing: bool = False,
         exit_first: bool = False,
     ) -> None:
         results = []
 
-        self.generate_base_venvs(
-            pattern,
-            recreate=recreate_venvs,
-            skip_deps=skip_base_install,
-            pythons=pythons,
-        )
+        # if not installed or build option is set to true
+        if not self.are_base_venvs_generated or build:
```

Comment:
I think we need to actively check here if the base venv exists or not.

Since `are_base_venvs_generated` will only be True after `generate_base_venvs` is called, but we don't save this state between CLI runs.

We want to support workflows like:


```
$ riot generate  # base venv is created
$ riot run redis  # base venv is not created
```

Or

```
$ rm -rf .riot
$ riot run redis  # base venv gets created
$ riot run redis  # base venv does not get created
```

Or

```
$ riot generate  # base venv is created
$ riot run --build redis  # base venv is recreated
```